### ISSUE TITLE
PlantUML can't be executed. Java missing?

### ISSUE THREAD

**Issue by danwos** [readthedocs/readthedocs.org#9958]
Since some days the execution of PlantUML ends with the below error.

The documentation itself hasn't changed and all configured folders are correct.

Is it possible that JAVA was removed or can't be used by simply calling `java`?

## Details

* Read the Docs project URL: https://readthedocs.org/projects/sphinx-needs/
* Build URL (if applicable): https://readthedocs.org/projects/sphinx-needs/builds/19268491/
* Read the Docs username (if applicable): https://readthedocs.org/profiles/danwos/

## Expected Result

PlantUML images get rentered, because JAVA can be used. 

## Actual Result

PlantUML diagrams get not rendered, following error appears in the log:
```
WARNING: plantuml command 'java -Djava.awt.headless=true -jar /home/docs/checkouts/readthedocs.org/user_builds/sphinx-needs/checkouts/stable/docs/utils/plantuml.jar' cannot be run
```

**Comment by humitos** [comment]
(I'm on the phone)

Have you changed the build image used? I saw you already commented on https://github.com/readthedocs/readthedocs.org/issues/3885 and you figured it out how to make it work some time ago.

You can install Java with build.apt_packes config option. Not that config is not supported on build.commands (there is another issue linked for that as well)

**Comment by danwos** [comment]
Thanks for the quick help.

Indeed, I (maybe) changed the build image ~2 weeks ago.
I needed to specify it to get Python 3.11 running, so it is now `ubuntu-22.04`.

I got java running by adding (thanks for the hint):
```
build:
  os: ubuntu-22.04
  apt_packages:
    - default-jdk
    - graphviz
  tools:
    python: "3.11"
```

But now I get this beautiful error image:
![image](https://user-images.githubusercontent.com/998700/215146041-0a9f0ba2-08e5-4125-a20d-5280b66ac495.png)

And I'm not able to solve it.
PlantUML docs are telling me, that PlantUML is looking into 2 folders for a `dot` executable:
![image](https://user-images.githubusercontent.com/998700/215146415-422ca26e-1615-4ee4-aa94-0fbe62266fc6.png)

`/usr/bin/dot` is the one that gets created when installing `graphviz`.
Tested via:
```
  jobs:
    pre_create_environment:
      - echo "DOT details"
      - which dot
      - export GRAPHVIZ_DOT=/usr/bin/dot
```
And as you see, I have set also an ENV var, which should tell PlantUML where to look for the `dot` executable.
But nothing works, I still get the above error.

I wasn't able to set a symbolic link between `/usr/bin/dot` and `/opt/local/bin/dot`, because `/opt/local/` does not exist and I don't have the rights (thanks god) to create folders on this level.

I tested this with PlantUML versions `2023.0` and `2022.14`. And I'm not sure if this may be also more of a PlantUML bug.

So I'm stuck and any ideas are welcome :) Thanks already for your help :+1:

**Comment by humitos** [comment]
(still on the phone)

Just wanted to note that environment variables defined in build.jobs are not exported to the following commands.

Please, try defining that variable using the project admin's page and let me know

**Comment by danwos** [comment]
Argh, it was already working with my above changes.
No need for setting the ENV var, but thanks for the hint.

My browser always showed a cached version, even after several reloads and over 1-2 hours.
Only after opening the developer tools, where caching is disabled, it loaded the rendered images. 

My final config now is:

1) Having a `plantuml.jar` under `docs/utils` in my repo.

2) Setting this in my `conf.py`:
```python
local_plantuml_path = os.path.join(os.path.dirname(__file__), "utils", "plantuml-1.2022.14.jar")
plantuml = f"java -Djava.awt.headless=true -jar {local_plantuml_path}"
```

3) And finally having this in my `.readthedocs.yml`:
```yml
build:
  os: ubuntu-22.04
  apt_packages:
    - default-jdk
    - graphviz

# and some other not-plantuml related stuff.
```

**Comment by humitos** [comment]
Excellent 👍

I'm happy to read this and that had documented your working solution in this thread. We should probably copy and officially document in our docs do other people can find it easily.

Thanks!

**Comment by jaygray0919** [comment]
Frickin' works!
Had spent hours fiddling with this.
Except for the pearl above, there is nearly zero useful documentation for Plant UML on RTD.
All hail @danwos !
